extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,319 @@
+  {
+    if (metricColumns == null) {
+      final BaseObjectColumnValueSelector<ArrayOfDoublesSketch> selector = metricFactory
+          .makeColumnValueSelector(fieldName);
+      if (selector instanceof NilColumnValueSelector) {
+        return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+      }
+      return new ArrayOfDoublesSketchMergeBufferAggregator(
+        selector,
+        nominalEntries,
+        numberOfValues,
+        getMaxIntermediateSize()
+      );
+    }
+    final DimensionSelector keySelector = metricFactory
+        .makeDimensionSelector(new DefaultDimensionSpec(fieldName, fieldName));
+    if (keySelector == null || DimensionSelectorUtils.isNilSelector(keySelector)) {
+      return new ArrayOfDoublesSketchNoOpBufferAggregator(numberOfValues);
+    }
+    final List<BaseDoubleColumnValueSelector> valueSelectors = new ArrayList<>();
+    for (final String column : metricColumns) {
+      final BaseDoubleColumnValueSelector valueSelector = metricFactory.makeColumnValueSelector(column);
+      valueSelectors.add(valueSelector);
+    }
+    return new ArrayOfDoublesSketchBuildBufferAggregator(
+      keySelector,
+      valueSelectors,
+      nominalEntries,
+      getMaxIntermediateSize()
+    );
+  }
+
+  @Override
+  public Object deserialize(final Object object)
+  {
+    return ArrayOfDoublesSketchOperations.deserialize(object);
+  }
+
+  @Override
+  public Comparator<ArrayOfDoublesSketch> getComparator()
+  {
+    return COMPARATOR;
+  }
+
+  @Override
+  public Object combine(final Object lhs, final Object rhs)
+  {
+    final ArrayOfDoublesUnion union = new ArrayOfDoublesSetOperationBuilder().setNominalEntries(nominalEntries)
+        .setNumberOfValues(numberOfValues).buildUnion();
+    updateUnion(union, lhs);
+    updateUnion(union, rhs);
+    return union.getResult();
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @JsonProperty
+  public int getNominalEntries()
+  {
+    return nominalEntries;
+  }
+
+  @JsonProperty
+  public List<String> getMetricColumns()
+  {
+    return metricColumns;
+  }
+
+  @JsonProperty
+  public int getNumberOfValues()
+  {
+    return numberOfValues;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Collections.singletonList(fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new CacheKeyBuilder(AggregatorUtil.ARRAY_OF_DOUBLES_SKETCH_CACHE_TYPE_ID).appendString(name)
+        .appendString(fieldName).appendInt(nominalEntries).appendStrings(metricColumns).appendInt(numberOfValues)
+        .build();
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return ArrayOfDoublesUnion.getMaxBytes(nominalEntries, numberOfValues);
+  }
+
+  public static final Comparator<ArrayOfDoublesSketch> COMPARATOR = new Comparator<ArrayOfDoublesSketch>()
```

Comment:
Move constant to the very beginning of the class declaration

Reply:
It's `Comparator.comparingDouble(ArrayOfDoublesSketch::getEstimate)`. And usually such comparators are null-aware in other impls, so wrap with `Comparator.nullsFirst(...)`